examples/hetero/hetero_unsupervised_dblp.py

```diff
@@ -0,0 +1,187 @@
+        edge_index_b = edge_index_dict[('paper', 'to', 'author')]
+
+        target_edge_index = \
+            self.convert_to_meta_path_edge_index(edge_index_a, edge_index_b)
+
+        pos_feat = target_embeds
+        # shuffle feature corruption
+        idxx = np.random.permutation(pos_feat.shape[0])
+        corrupted_feat = pos_feat[idxx]
+
+        pos_embed = self.encoder(pos_feat, target_edge_index)
+
+        neg_embed = self.encoder(corrupted_feat, target_edge_index)
+
+        summary = torch.mean(pos_embed, dim=0)
+
+        return pos_embed, neg_embed, summary
+
+    def convert_to_meta_path_edge_index(self, edge_index_a, edge_index_b):
+        paper2author_dict = \
+            {idx_a.item(): idx_b.item() for idx_a, idx_b in edge_index_b.T}
+        target_edge_index = []
+        for idx_a, idx_b in edge_index_a.T:
+            idx_b = idx_b.item()
+            idx_a = idx_a.item()
+            if paper2author_dict.get(idx_b) is not None:
+                target_edge_index.append((idx_a, paper2author_dict[idx_b]))
+        target_edge_index = torch.tensor(target_edge_index, dtype=torch.long)
+        target_edge_index = target_edge_index.T
+
+        return target_edge_index
+
+    def discriminate(self, z, summary, sigmoid=True):
+        value = torch.matmul(z, torch.matmul(self.weight, summary))
+        return torch.sigmoid(value) if sigmoid else value
+
+    def reset_parameters(self):
+        uniform(self.embed_size, self.weight)
+
+    def loss(self, pos_embed, neg_embed, summary):
+
+        pos_loss = -torch.log(
+            self.discriminate(pos_embed, summary, sigmoid=True) + EPS
+        ).mean()
+        neg_loss = -torch.log(
+            1 - self.discriminate(neg_embed, summary, sigmoid=True) + EPS
+        ).mean()
+
+        return pos_loss + neg_loss
+
+
+model = HeteroUnsupervised(
+    data.metadata(),
+    out_channels=64,
+    hidden_channels=64,
+    num_layers=2
+)
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+data, model = data.to(device), model.to(device)
+
+with torch.no_grad():  # Initialize lazy modules.
+    out = model(data.x_dict, data.edge_index_dict)
+
+optimizer = torch.optim.Adam(model.parameters(),
+                             lr=0.005, weight_decay=0.001)
+
+
+def train():
+    model.train()
+    optimizer.zero_grad()
+    pos_embed, neg_embed, summary = model(data.x_dict, data.edge_index_dict)
+
+    # pick out the train embed
+    mask = data['author']['train_mask']
+    pos_embed = pos_embed[mask]
+    neg_embed = neg_embed[mask]
+
+    loss = model.loss(pos_embed, neg_embed, summary)
+    loss.backward()
+    optimizer.step()
+    return float(loss)
+
+
+@torch.no_grad()
+def test():
+    model.eval()
+    pos_embed, _, _ = model(data.x_dict, data.edge_index_dict)
+    mask = data['author']['val_mask']
+    labels = data['author'].y[mask]
+
+    # I dont get it why it dosen't work
+    valid_embed = pos_embed[mask]
```

Comment:
What does not work?

Reply:
Please ignore this annotation, I forgot to delete it lol....

Reply:
Ok, just let me know once you feel this PR is ready for review :)